pkg/apiserver/resthandler.go

```diff
@@ -269,8 +269,14 @@ func CreateResource(r rest.Creater, scope RequestScope, typer runtime.ObjectType
 			errorJSON(err, scope.Codec, w)
 			return
 		}
+		// Get name, ok if empty
+		_, name, _ := scope.Namer.Name(req)
+
 		ctx := scope.ContextFunc(req)
 		ctx = api.WithNamespace(ctx, namespace)
+		if len(name) > 0 {
+			ctx = api.WithValue(ctx, "name", name)
```

Comment:
If we set the name on the object we may be hiding an error condition. You may be posting to a particular path and post an object with a different name. I would think you'd want the server to tell you that you have a mismatch instead of just taking the name from the path.